pycoin/ecdsa/ellipticcurve.py

```diff
@@ -67,7 +72,8 @@ def __init__( self, curve, x, y, order = None ):
     self.__y = y
     self.__order = order
     # self.curve is allowed to be None only for INFINITY:
-    if self.__curve: assert self.__curve.contains_point( x, y )
+    if self.__curve and not self.__curve.contains_point( x, y ):
+      raise ValueError('({},{}) is not on the curve {}'.format(x, y, curve))
```

Comment:
Using a pycoin-specific exception here would be nice. ValueError is too general for others to catch and know what's going on.